js/foam/browser/u2/BrowserView.js

```diff
@@ -0,0 +1,450 @@
+                  this.data.headerColor + '; }')
+              .end();
+          var header = this.start('div')
+              .x({ data: this })
+              .cls(this.myCls('header'))
+              .cls(this.myCls('header-color'))
+              .cls2(function(sm) {
+                return sm && 'foam-u2-Element-hidden';
+              }.on$(this.X, this.searchMode$));
+
+          header.add(this.MENU_BUTTON);
+          header.start('span')
+              .cls(this.myCls('header-title'))
+              .add(this.title$)
+              .end();
+
+          if (this.spinner) header.add(this.spinner);
+
+          // TODO(braden): Parent actions. How to create them with the right
+          // data?
+
+          header.add(this.SEARCH_BUTTON);
+
+          // TODO(braden): Sort order drop-down menu.
+          header.end();
+
+          var search = this.start('div')
+              .x({ data: this })
+              .cls(this.myCls('header'))
+              .cls(this.myCls('search-header'))
+              .cls(this.myCls('header-color'))
+              .cls2(function(sm) {
+                return sm || 'foam-u2-Element-hidden';
+              }.on$(this.X, this.searchMode$))
+              .add(this.BACK_BUTTON)
+              .add(this.data.SEARCH);
+          search.end();
+
+          // This is needed to make sure the input box gets focused in the same
+          // user input event that triggered the mode switch; that will cause
+          // the keyboard to appear on mobile.
+          this.searchMode$.addListener(function() {
+            if (this.searchMode) {
+              search.children[1].focus();
+            }
+          }.bind(this));
+
+          this.listView_ = this.data.listView({
+            data$: this.data.filteredDAO$
+          }, this.Y);
+          this.start('div').cls(this.myCls('body')).add(this.listView_).end();
+
+          if (this.data.showAdd) {
+            // TODO(braden): Make this a floating button, somehow.
+            var button = this.CREATE_BUTTON.toE(this.Y);
+            button.type = 'floating';
+            button.color = '#fff';
+            button.data = this;
+            this.add(button);
+          }
+
+          this.menuView_ = this.start('div')
+              .cls(this.myCls('menu-container'))
+              .cls2(function(open) {
+                return open && self.myCls('menu-open');
+              }.on$(this.X, this.menuOpen$))
+              .start('div').cls(this.myCls('menu-inner'))
+                  .start('div').cls(this.myCls('menu-overlay'))
+                      .on('click', this.onMenuTouch)
+                      .end()
+                  .start('div').cls(this.myCls('menu-body')).end()
+              .end();
+          this.menuView_.end();
+        },
+      ],
+    }
+  ],
+
+  properties: [
+    {
+      name: 'touchManager',
+      factory: function() {
+        return this.X.touchManager || this.TouchManager.create();
```

Comment:
Instead of factories like this you can import touchManager, and just leave the factory as return this.TouchManager.create().  The factory won't run if the import sets the value.